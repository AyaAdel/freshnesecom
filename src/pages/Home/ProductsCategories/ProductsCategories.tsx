import ListGroup from 'components/ListGroup/ListGroup';
import { getProductsOfCategory } from 'services/products';
import { useAppSelector, useAppDispatch } from 'app/hooks';
import type { RootState, AppDispatch } from 'app/store';
import { loadProducts } from 'features/products/productsSlice';

export default function ProductsCategories() {
  const categories: string[] = useAppSelector(
    (state: RootState) => state.categories
  );

  const dispatch: AppDispatch = useAppDispatch();

  const filterProducts = async (category: string) => {
    const response = await getProductsOfCategory(category);
    dispatch(loadProducts(response));
  };

  return (
    <ListGroup title="Categories" links={categories} onClick={filterProducts} />
  );
}
