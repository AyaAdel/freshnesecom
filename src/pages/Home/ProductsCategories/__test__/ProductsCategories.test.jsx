import React from 'react';
import renderer from 'react-test-renderer';
import { RenderComponent } from 'utils/test/render';
import ProductsCategories from '../ProductsCategories';

function MyComponent() {
  return (
    <RenderComponent>
      <ProductsCategories />
    </RenderComponent>
  );
}

it('Render Products Categories Correctly', () => {
  const tree = renderer.create(<MyComponent />).toJSON();
  expect(tree).toMatchSnapshot();
});
