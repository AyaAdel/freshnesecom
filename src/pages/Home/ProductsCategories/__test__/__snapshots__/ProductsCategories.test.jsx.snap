// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Products Categories Correctly 1`] = `
<div
  className="list-group"
>
  <h4
    className="list-group__title"
  >
    Categories
  </h4>
  <div
    className="list-group__links"
  />
</div>
`;
